Cookbook/Proposals/SimpleSideEffectInReactiveFeedback.md

```diff
@@ -0,0 +1,124 @@
+
+A typical implementation of analytics event goes like this:
+
+```swift
+class ViewModel {
+    let (userAction, input) = Signal.pipe()
+    let state: Property<State>
+    let routes: Signal<Route>
+
+    init(analytics: Analytics, ...) {
+        analytics.track(TrackingEvent.initialized)
+
+        /* ReactiveFeedback setup */
+        self.state = Property(
+            ...,
+            feedbacks: [
+                whenButtonTapped(analytics: analytics)
+            ]
+        )
+
+        self.routes = state.signal
+            .filterMap { $0.route }
+            .on(value: { route in
+                analytics.track(TrackingEvent.routingByFeedback(route))
+            })
+
+        userAction
+            .filterMap(Route.init(userAction:))
+            .observeValues { route in
+                analytics.track(TrackingEvent.routingByUser(route))
+            }
+    }
+
+    func send(action: UserAction) {
+        input.observer(Event.ui(action))
+        analytics.track(TrackingEvent.userAction(action)
+    }
+}
+
+// MARK: - Feedbacks
+
+func whenButtonTapped(analytics: Analytics) -> Feedback {
+    return Feedback { signal in
+        return signal
+            .flatMap(.latest) { someAPI }
+            .on(value: { response in
+                analytics.track(TrackingEvent.apiComplete(response))
+            })
+            .map(Event.didFinishAPI)
+    }
+}
+```
+
+Please notice **how `analytics.track()`s are scattered across `ViewModel`**, mostly outside of `RAF`'s feedback system.
+This is because FRP e.g. `ReactiveSwift` is **too easy to inject side-effects anywhere** by using `on(value:)` and `observeValues`.
+While maintaining these side-effects inside `ViewModel` (rather than `RAF`) could be sufficient, "where to implement side-effects" ambiguity still remains, which is not ideal for practicing `RAF` architecture.
+
+In contrast, other architectures e.g. [Elm](https://elm-lang.org) and [Redux](https://redux.js.org) focuses their side-effect handling in `Feedback` layer (`Cmd` and `Middleware`) that _may_ send next feedback event.
+It's worth noting that **non-feedback type `Feedback<Never>` is a subset of `Feedback<Event>`**, thus thinking `Feedback` type as a side-effect handler is more important than using it as a feedback loop.
+
+## Proposed solution
+
+This proposal will let us handle side-effects inside `Feedback` without actual feedback loop implemented.
+
+For example, instead of writing:
+
+```swift
+    func send(action: UserAction) {
+        ...
+        analytics.track(TrackingEvent.userAction(action)
+    }
+```
+
+we can replace with `Feedback`:
+
+```swift
+let whenUserActionReceived = Feedback.noFeedback(^\Event.userAction) {
+    analytics.track(TrackingEvent.userAction($0))
+}
+
+extension Feedback {
+    /// Helper: simple side-effect without feedback-loop.
+    static func noFeedback<Value>(_ filter: Event -> Value?, _ f: Value -> Void) -> Signal<Event> {
```

Comment:
Maybe `static func sideEffect`? 
However `Feedback` is a kind of side effect already 🤔.  For me it's still more verbose than `noFeedback` tho

Reply:
Should we maybe explore some functional concepts for this, like `IO`? (if it's completely unrelated then ignore this ^^)